vmware-event-router/internal/provider/vcenter/checkpoint.go

```diff
@@ -126,19 +127,21 @@ func lastCheckpoint(_ context.Context, file io.Reader) (*checkpoint, error) {
 // createCheckpoint creates a checkpoint using the given file, vcenter host name
 // and checkpoint timestamp returning the created checkpoint. If lastEvent is
 // nil an errInvalidEvent will be returned.
-func createCheckpoint(_ context.Context, file io.Writer, vcHost string, lastEvent types.BaseEvent, timestamp time.Time) (*checkpoint, error) {
-	// won't catch the case when the value is not pointer
-	if lastEvent == nil || reflect.ValueOf(lastEvent).IsNil() {
+func createCheckpoint(_ context.Context, file io.Writer, vcHost string, last lastEvent, timestamp time.Time) (*checkpoint, error) {
+	be := last.baseEvent
+
+	// will panic when the baseEvent value is not pointer
```

Comment:
This is a clearer comment than the previous one. Good change.

Reply:
Yeah, your feedback was very valuable to be more descriptive and not assume it's obvious.